src/Decoder.c

```diff
@@ -4172,16 +4417,34 @@ static ZyanStatus ZydisPopulateRegisterIds(ZydisDecoderContext* context,
         const ZyanBool is_emvex = (instruction->encoding == ZYDIS_INSTRUCTION_ENCODING_EVEX) ||
```

Comment:
This function needs to be revisited. For CR and DR the high 16 register are invalid. For other types they are ignored.